pkg/controller/shared_state.go

```diff
@@ -1564,3 +1571,60 @@ func (c *SharedState) getNodesTopologySelector(ctx context.Context, nodes []v1.N
 	}
 	return topologySelector
 }
+
+func (s *SharedState) addNodeToAvailableAttachmentsMap(ctx context.Context, node string) {
+	nodeObj := &v1.Node{}
+	if err := s.cachedClient.Get(ctx, types.NamespacedName{Name: node}, nodeObj); err != nil {
+		capacity, _ := azureutils.GetNodeMaxDiskCountWithLabels(nodeObj.Labels)
+		var count atomic.Int32
+		count.Store(int32(capacity))
+		s.availableAttachmentsMap.Store(node, &count)
+		return
+	}
+	capacity, _ := azureutils.GetNodeRemainingDiskCount(ctx, s.cachedClient, node)
+	var count atomic.Int32
+	count.Store(int32(capacity))
+	s.availableAttachmentsMap.Store(node, &count)
+}
+
+func (s *SharedState) deleteNodeFromAvailableAttachmentsMap(ctx context.Context, node string) {
+	w, _ := workflow.GetWorkflowFromContext(ctx)
+	w.Logger().V(5).Infof("Delete node (%s) from availableAttachmentsMap", node)
```

Comment:
Rather the log a formatted string, use `Info` and pass the node as a key value pair, e.g.

```go
w.Logger().V(5).Info("Deleted node from availableAttachmentsMap", consts.NodeNameLabel, node)
```

Reply:
What's the benefit of using Info() rather than Infof()?

Reply:
It uses structured logging which is easier to parse for automated analysis.

Reply:
But Info() can only have 1 argument (see bottom). You must mean using infof() right?
w.Logger().V(5).Infof("Deleted node from availableAttachmentsMap", consts.NodeNameLabel, node)

```
func (l Logger) Info(msg string) {
	l.WithCallDepth(1).Info(msg)
}
```

Reply:
Ah, my bad. I forgot that the workflow logger interface does not implement the additional `keysAndValues` parameters that the [`logr.Logger.Error`](https://pkg.go.dev/github.com/go-logr/logr#Logger.Error) method does.